pkg/monitortests/testframework/watchevents/monitortest.go

```diff
@@ -0,0 +1,55 @@
+package watchevents
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitortestframework"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type eventWatcher struct {
+}
+
+func NewEventWatcher() monitortestframework.MonitorTest {
+	return &eventWatcher{}
+}
+
+func (w *eventWatcher) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	kubeClient, err := kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+
+	startEventMonitoring(ctx, recorder, adminRESTConfig, kubeClient)
+
+	return nil
+}
+
+func (w *eventWatcher) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	// because we are sharing a recorder that we're streaming into, we don't need to have a separate data collection step.
```

Comment:
I'm curious if you're certain we need three ways a monitortest can inject intervals, CollectData, ConstructComputedIntervals (both of which make sense), but why do we need live recording as well? On the surface it feels like that could be in CollectData? 

Depending on the rationale it might be helpful to spell it out in the monitortest README when to use each.

Reply:
There isn't any ordering, so the gating is required to have type A CollectData for Calculators 1 and 2

Live recording handles th e watches.